Source/Core/VideoCommon/PerformanceTracker.cpp

```diff
@@ -38,193 +36,155 @@ PerformanceTracker::~PerformanceTracker()
 
 void PerformanceTracker::Reset()
 {
-  std::unique_lock lock{m_mutex};
+  m_raw_dts.Clear();
+  m_dt_queue.clear();
 
-  QueueClear();
+  m_dt_total = DT::zero();
+  m_last_raw_dt = DT::zero();
   m_last_time = Clock::now();
   m_hz_avg = 0.0;
   m_dt_avg = DT::zero();
-  m_dt_std = std::nullopt;
+  m_dt_std = DT::zero();
+  m_is_last_time_sane = false;
 }
 
 void PerformanceTracker::Count()
 {
-  std::unique_lock lock{m_mutex};
+  const TimePoint current_time{Clock::now()};
 
-  if (m_paused)
+  const DT diff{current_time - m_last_time};
+  m_last_time = current_time;
+
+  if (!m_is_last_time_sane)
+  {
+    m_is_last_time_sane = true;
     return;
+  }
 
-  const DT window{GetSampleWindow()};
+  m_last_raw_dt = diff;
+  m_raw_dts.Push(diff);
+}
 
-  const TimePoint time{Clock::now()};
-  const DT diff{time - m_last_time};
+void PerformanceTracker::UpdateStats()
+{
+  DT diff{};
+  while (m_raw_dts.Pop(diff))
+    HandleRawDt(diff);
+}
 
-  m_last_time = time;
+void PerformanceTracker::HandleRawDt(DT diff)
+{
+  if (m_dt_queue.size() == MAX_DT_QUEUE_SIZE)
+    PopBack();
 
-  QueuePush(diff);
-  m_dt_total += diff;
+  PushFront(diff);
 
-  if (m_dt_queue_begin == m_dt_queue_end)
-    m_dt_total -= QueuePop();
+  const DT window{GetSampleWindow()};
 
-  while (window <= m_dt_total - QueueTop())
-    m_dt_total -= QueuePop();
+  while (m_dt_total - m_dt_queue.back() >= window)
+    PopBack();
 
   // Simple Moving Average Throughout the Window
-  m_dt_avg = m_dt_total / QueueSize();
-  const double hz = DT_s(1.0) / m_dt_avg;
+  const DT dt_avg = m_dt_total / m_dt_queue.size();
+  const double hz = DT_s(1.0) / dt_avg;
+  m_dt_avg = dt_avg;
 
   // Exponential Moving Average
   const DT_s rc = SAMPLE_RC_RATIO * std::min(window, m_dt_total);
   const double a = 1.0 - std::exp(-(DT_s(diff) / rc));
 
   // Sometimes euler averages can break when the average is inf/nan
-  if (std::isfinite(m_hz_avg))
-    m_hz_avg += a * (hz - m_hz_avg);
+  const auto hz_avg = m_hz_avg.load();
+  if (std::isfinite(hz_avg))
+    m_hz_avg = hz_avg + a * (hz - hz_avg);
   else
     m_hz_avg = hz;
 
-  m_dt_std = std::nullopt;
+  // Update Std Dev
+  MathUtil::RunningVariance<double> variance;
+  for (auto dt : m_dt_queue)
+    variance.Push(DT_s(dt).count());
+  m_dt_std = std::chrono::duration_cast<DT>(DT_s(variance.PopulationStandardDeviation()));
```

Comment:
You probably only want to update the standard deviation on `UpdateStats` and not `HandleRawDT` right?